build: fix compilation with meson 1.3.0

ret2libc opened the pull request:
# DO NOT SQUASH

ret2libc committed:
build: fix compilation with meson 1.3.0

XVilka reviewed (changes requested):
Looks like it broke unit tests on ASAN for some reason:
```
  63/101 rizin:unit / intervaltree         FAIL            0.31s   killed by signal 6 SIGABRT
>>> MALLOC_PERTURB_=136 UBSAN_OPTIONS=halt_on_error=1:abort_on_error=1:print_summary=1:print_stacktrace=1 LD_LIBRARY_PATH=/home/runner/work/rizin/rizin/build/librz/crypto:/home/runner/work/rizin/rizin/build/librz/analysis:/home/runner/work/rizin/rizin/build/librz/debug:/home/runner/work/rizin/rizin/build/librz/socket:/home/runner/work/rizin/rizin/build/librz/cons:/home/runner/work/rizin/rizin/build/librz/il:/home/runner/work/rizin/rizin/build/librz/hash:/home/runner/work/rizin/rizin/build/librz/demangler:/home/runner/work/rizin/rizin/build/librz/egg:/home/runner/work/rizin/rizin/build/librz/io:/home/runner/work/rizin/rizin/build/librz/magic:/home/runner/work/rizin/rizin/build/librz/search:/home/runner/work/rizin/rizin/build/librz/core:/home/runner/work/rizin/rizin/build/librz/parse:/home/runner/work/rizin/rizin/build/librz/syscall:/home/runner/work/rizin/rizin/build/librz/sign:/home/runner/work/rizin/rizin/build/librz/flag:/home/runner/work/rizin/rizin/build/librz/bp:/home/runner/work/rizin/rizin/build/librz/diff:/home/runner/work/rizin/rizin/build/librz/type:/home/runner/work/rizin/rizin/build/librz/lang:/home/runner/work/rizin/rizin/build/librz/config:/home/runner/work/rizin/rizin/build/librz/bin:/home/runner/work/rizin/rizin/build/librz/main:/home/runner/work/rizin/rizin/build/librz/reg:/home/runner/work/rizin/rizin/build/librz/util:/home/runner/work/rizin/rizin/build/librz/asm:/usr/lib/llvm-14/lib/clang/14.0.0/lib/linux:/home/runner/lib/x86_64-linux-gnu:/home/runner/lib:/home/runner/lib64: /home/runner/work/rizin/rizin/build/test/unit/test_intervaltree
```

ret2libc committed:
Revert "ci: downgrade Meson to 1.2.3 (#3991)"

This reverts commit 6aeea82800a253f0c4f6b6470750ab85106b5579.

ret2libc committed:
Revert "Downgrade Docker Meson to 1.2.3 (#3994)"

This reverts commit 94d7ccc75218de6daf209c6b2309ae29d53e7a4e.

XVilka reviewed (changes requested):
```
[libzip] configuring '/home/runner/work/rizin/rizin/build/subprojects/libzip-1.9.2/config.h'
/home/runner/work/rizin/rizin/subprojects/libzip-1.9.2/cmake-config.h.in:8:30: error cmakedefine only supports two modes:
	--> #cmakedefine HAVE___PROGNAME
 	--> #cmakedefine HAVE___PROGNAME @HAVE___PROGNAME@
In particular, no value other than "HAVE___PROGNAME" will be substituted and
no extra characters (comments, etc.) are allowed on the line.
  8 | #cmakedefine HAVE___PROGNAME ${HAVE___PROGNAME}
                                   ^
/home/runner/work/rizin/rizin/subprojects/libzip-1.9.2/meson.build:114:12: error in function configure_file()
114 | config_h = configure_file(
                 ^
/home/runner/work/rizin/rizin/meson.build:575:17: error in function subproject()
575 |   libzip_proj = subproject('libzip', default_options: [
                      ^
Error: Process completed with exit code 1.
```

ret2libc committed:
ci: use default meson even in create-tarball

ret2libc committed:
test: temporarily disable test_rz_analysis_var

There is an issue when running it under UBSAN, however this issue has
always been there but it presents itself only with some specific values
of MALLOC_PERTURB. We are temporarily disabling this because we need to
fix an issue with meson-1.3.0 so that people can easily compile the
project.

ret2libc commented:
```
>>> ASAN_OPTIONS=halt_on_error=1:abort_on_error=1:print_summary=1 MALLOC_PERTURB_=252 UBSAN_OPTIONS=halt_on_error=1:abort_on_error=1:print_summary=1:print_stacktrace=1 /__w/rizin/rizin/rizin/build/test/unit/test_subprocess
――――――――――――――――――――――――――――――――――――― ✀  ―――――――――――――――――――――――――――――――――――――
stdout:
test_noargs_noinput_outerr OK
Hello World
Hello World
test_args OK
test_env OK
test_stdin OK
test_multi OK
test_specialchar_args OK
test_nopipes OK
test_stdoutonly OK
test_stderronly OK
test_stdoutstderr OK
test_interactive OK
test_interactive_pty OK
test_interactive_custom_pty OK
test_interactive_not_raw_pty ERR
[XX] Fail at line 367: No 3 + 5 + c: expected 2072314[104](https://github.com/rizinorg/rizin/actions/runs/7059979643/job/19218680371?pr=3995#step:7:105), got 2072.

stderr:
This is on err
This is on err
```


I saw this failing also on separate PR :( i'd be for merging this anyway and fix this before release.
